contracts-periphery/script/Deploy.s.sol

```diff
@@ -1,15 +1,29 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.16;
 
-import "forge-std/Script.sol";
+import {Script} from "forge-std/Script.sol";
+import {StdChains} from "forge-std/StdChains.sol";
 import {UmbraBatchSend} from "src/UmbraBatchSend.sol";
 import {IUmbra} from "src/interface/IUmbra.sol";
 
 contract Deploy is Script {
   address constant UMBRA = 0xFb2dc580Eed955B528407b4d36FfaFe3da685401;
 
-  function run() public {
+  function deploy() internal {
     vm.broadcast();
     new UmbraBatchSend(IUmbra(UMBRA));
   }
 }
+
+contract ScriptBase is Deploy {
```

Comment:
I think you have things flipped here? `contract Deploy is ScriptBase` would be more logical. If we plan to use the base script for other deploy scripts too, let's pull it out into a separate file that does not have the `.s` extension, e.g. `script/BaseScript.sol`

Reply:
ohhh hah yes that def makes more sense

Reply:
Like this https://github.com/ScopeLift/umbra-protocol/pull/504/commits/d5716543b4c003378cfa32abf56ed0935dac6a69 ?

Reply:
Not exactly, sorry let me clarify:
- `Base*` typically means that thing (whether a contract, class, etc) is meant to be inherited from and used at the base for other things
- In the forge scripting case, a `BaseScript` would contain methods, tools, or utilities that you want all your forge scripts to share
- What are those in this case? To me, it seems like it's (1) calling `setFallbackToDefaultRpcUrls(false)`, and (2) checking if a script is already deployed and deploying it if not
- Based on that logic, the current architecture swaps the content of what should be in the deploy script and the base script: `BaseScript` has the deploy method for the batch send contract which doesn't make much sense IMO

However, both of these scripts are simple enough, that we're probably over-abstracting. Let's instead create a new file called `DeployBatchSend.s.sol` and give the contract the same name. This makes it clear what this file does, where as the current file names are ambiguous—what is `Deploy.s.sol` actually deploying? Then we can put all code for this script into the `run` method of that contract

Reply:
Thanks for the clarification! Ik something didn't feel right cause the base contract seemed too specific to `batchSend` in this case, but your explanation made it clear. I've merged the scripts for now with the `DeployBatchSend.s.sol` name here: https://github.com/ScopeLift/umbra-protocol/commit/9bfbf40edf11a8894d7efadeb07a8b313fee229a